extra/js/URLSearchParams.js

```diff
@@ -0,0 +1,45 @@
+function getURLParams() {
+    return new Proxy(new URLSearchParams(window.location.search), {
+        get: (searchParams, prop) => searchParams.get(prop),
+    });
+}
+
+// from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/decodeURIComponent
+function containsEncodedComponents(x) {
+  // ie ?,=,&,/ etc
+  return (decodeURI(x) !== decodeURIComponent(x));
+}
+
+function injectSearchfromURL(searchToken = "searchText") {
```

Comment:
is search token a URL decoded string? Maybe let's add a docstring

Reply:
`searchToken = "searchText"` just defines the query key, for instance, `docs.mat3ra.com/?searchText=Property`. The keys and values are automatically grabbed by `URLSearchParams()`.